pgmpy/estimators/HillClimbSearch.py

```diff
@@ -283,24 +272,33 @@ def estimate(
                 "'start_dag' should be a DAG with the same variables as the data set, or 'None'."
             )
 
-        # Step 1.3: Check fixed_edges
-        if not hasattr(fixed_edges, "__iter__"):
-            raise ValueError("fixed_edges must be an iterable")
+        if expert_knowledge:
+            # Step 1.3: Check fixed_edges
+            if expert_knowledge.fixed_edges == set():
+                fixed_edges = set()
+            else:
+                fixed_edges = set(expert_knowledge.fixed_edges)
+                start_dag.add_edges_from(fixed_edges)
+                if not nx.is_directed_acyclic_graph(start_dag):
+                    raise ValueError(
+                        "fixed_edges creates a cycle in start_dag. Please modify either fixed_edges or start_dag."
+                    )
+
+            # Step 1.4: Check black list and white list
+            black_list = (
```

Comment:
These prepocessing steps can be moved to  `ExpertKnowlege._validate_edges` ?

Reply:
Can be done for black list and fixed edges. For white edges, we cannot move it because because we need the variables of dataset to build the white list in the case when the user does not provide it. Unless, as you suggested we get rid of building a white list set (when the user does not provide one) and simply make it None.

Reply:
Yes, I think it's a better idea to keep white_list as None.

Reply:
Do you think instead of keeping white_list as None, we just keep it as an empty set like the fixed edges and blacklist? Won't be an issue, right? Just would look more consistent I think.

Reply:
@ankurankan 

> Do you think instead of keeping white_list as None, we just keep it as an empty set like the fixed edges and blacklist? Won't be an issue, right? Just would look more consistent I think.

Reply:
@Nimish-4 I think setting it to empty set would be bit confusing for the user because an empty set would imply that all edges are possible.